modules/platforms/dotnet/Apache.Ignite.Core.Tests/Common/IgniteProductVersionTests.cs

```diff
@@ -0,0 +1,158 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Apache.Ignite.Core.Tests.Common
+{
+    using System;
+    using System.IO;
+    using System.Reflection;
+    using Apache.Ignite.Core.Binary;
+    using Apache.Ignite.Core.Common;
+    using Apache.Ignite.Core.Impl.Binary;
+    using Apache.Ignite.Core.Impl.Binary.IO;
+    using NUnit.Framework;
+
+    /// <summary>
+    /// Tests the <see cref="IgniteProductVersion"/>.
+    /// </summary>
+    public class IgniteProductVersionTests
+    {
+        private static readonly byte[] ByteArray = {24, 229, 167, 236, 158, 50, 2, 18, 106, 105, 188, 35, 26, 107, 150, 91, 193, 215, 61, 238};
+
+        private IgniteProductVersion _defaultVersion;
+
+        [SetUp]
+        public void SetUp()
+        {
+            var timeStamp = (new DateTime(2018, 1, 1).Ticks - BinaryUtils.JavaDateTicks) / 1000;
+            _defaultVersion = new IgniteProductVersion(2, 5, 7, timeStamp, null);
+        }
+
+        [Test]
+        public void TestBinaryConstructor()
+        {
+            IBinaryRawReader reader = SetUpRawBinaryReader(_defaultVersion.RevisionHash);
+
+            IgniteProductVersion deserializedVersion = new IgniteProductVersion(reader);
+
+            Assert.AreEqual(_defaultVersion, deserializedVersion);
+        }
+
+        private IBinaryRawReader SetUpRawBinaryReader(byte[] hash)
+        {
+            var marsh = BinaryUtils.Marshaller;
+
+            using (var stream = new BinaryHeapStream(128))
+            {
+                var writer = marsh.StartMarshal(stream);
+
+                writer.WriteByte(_defaultVersion.Major);
+                writer.WriteByte(_defaultVersion.Minor);
+                writer.WriteByte(_defaultVersion.Maintenance);
+                writer.WriteLong(_defaultVersion.RevisionTimestamp);
+                writer.WriteByteArray(hash);
+
+                stream.Seek(0, SeekOrigin.Begin);
+
+                return marsh.StartUnmarshal(stream);
+            }
+        }
+
+        [Test]
+        public void TestEqualityWithNull()
+        {
+            Assert.IsFalse(_defaultVersion.Equals(null));
+        }
+
+        [Test]
+        public void TestToString()
+        {
+            Assert.AreEqual("2.5.7#20180101-sha1:", _defaultVersion.ToString());
+        }
+
+        [Test]
+        public void TestToStringWithLimitedHashValue()
+        {
+            var hash = new byte[] {24, 229};
+            IBinaryRawReader reader = SetUpRawBinaryReader(hash);
+
+            IgniteProductVersion deserializedVersion = new IgniteProductVersion(reader);
+            Assert.AreEqual("2.5.7#20180101-sha1:18", deserializedVersion.ToString());
+        }
+
+        [Test]
+        public void TestLocalNodeGetVersion()
+        {
+            using (IIgnite ignite = Ignition.Start(TestUtils.GetTestConfiguration()))
```

Comment:
`Ignition.Start` is expensive. Please use `[TestFixtureSetUp]` to start a node once for all tests, or derive from `TestBase`.